src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeOptionalFieldsNode.cs

```diff
@@ -53,10 +57,31 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             if (!relocsOnly)
             {
                 _owner.ComputeOptionalEETypeFields(factory, relocsOnly: false);
-                objData.EmitBytes(_owner.GetOptionalFieldsData(factory));
+                objData.EmitBytes(_owner.GetOptionalFieldsData());
             }
             
             return objData.ToObjectData();
         }
+
+        protected override int ClassCode => (int)ObjectNodeOrder.EETypeOptionalFieldsNode;
+
+        protected override int CompareToImpl(ObjectNode other, TypeSystemComparer comparer, NameMangler nameMangler)
+        {
+            var compare = comparer.Compare(_owner.Type, ((EETypeOptionalFieldsNode)other)._owner.Type);
```

Comment:
Why do we need this complicated logic instead of just delegating to EETypeNode to sort it out?